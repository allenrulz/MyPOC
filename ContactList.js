import React,{Component} from 'react';
import { StyleSheet, Text, View,FlatList } from 'react-native';
import customData from '../Data/MOCK_DATA.json';


export default class Landing extends React.Component {

  constructor(props){
    super(props);
    //console.log("testing");
    //console.log({this.state.names});
    this.state = {names: ( typeof this.props.navigation.state === 'undefined' ? this.props.names : this.props.navigation.state.params.name)}
  }

  static navigationOptions = {
    title: 'List',
  };

  contactNames() {
    //firstnames = customData.map(a => ['first_name': a.first_name]);
    //console.log(this.state.names);

    newList = customData.filter(obj => obj.first_name.includes(this.state.names));

    console.log(newList);


    return(
      <FlatList
        data={newList}
        //data={[{key: 'a'}, {key: 'b'}]}
        renderItem={({item}) => <Text>{item.first_name}</Text>}
/>
    );

}



  render() {
    const names  = this.props.names;
    console.log(names);
    const{ navigate } = this.props.navigation;
    return (
      <View>
        {this.contactNames()}
        </View>


      //{console.log((customData[2]['Company']))}



    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
