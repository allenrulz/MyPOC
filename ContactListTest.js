import React from 'react';
import ContactList from '../ContactList';

import renderer from 'react-test-renderer'

const navigation = { navigate: jest.fn() };
const names = jest.fn().mockReturnValue('All');

//this.setState({params.name : 'all'});

test('renders correctly with All', () => {
  const tree = renderer.create(<ContactList navigation={navigation} names='Dia'/>).toJSON();
  expect (tree).toMatchSnapshot();

});

test('renders correctly with Dia', () => {
  const tree = renderer.create(<ContactList navigation={navigation} names='All'/>).toJSON();
  expect (tree).toMatchSnapshot();

});
