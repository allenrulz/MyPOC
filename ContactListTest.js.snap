// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with All 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "Company": "Devcast",
          "Title": "Mechanical Systems Engineer",
          "email": "delie81@washingtonpost.com",
          "first_name": "Diane-marie",
          "gender": "Female",
          "id": 290,
          "last_name": "Elie",
        },
        Object {
          "Company": "Youspan",
          "Title": "Information Systems Manager",
          "email": "dlangstonejv@bloglines.com",
          "first_name": "Diandra",
          "gender": "Female",
          "id": 716,
          "last_name": "Langstone",
        },
        Object {
          "Company": "Lazz",
          "Title": "Chemical Engineer",
          "email": "dforankq@whitehouse.gov",
          "first_name": "Diarmid",
          "gender": "Male",
          "id": 747,
          "last_name": "Foran",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Diane-marie
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Diandra
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Diarmid
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly with Dia 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "Company": "Jetpulse",
          "Title": "Design Engineer",
          "email": "ariallbf@nationalgeographic.com",
          "first_name": "Alley",
          "gender": "Male",
          "id": 412,
          "last_name": "Riall",
        },
        Object {
          "Company": "Zoombeat",
          "Title": "Social Worker",
          "email": "aschusterlh9@cafepress.com",
          "first_name": "Allen",
          "gender": "Male",
          "id": 622,
          "last_name": "Schusterl",
        },
        Object {
          "Company": "Wikido",
          "Title": "Director of Sales",
          "email": "abryentoniz@theatlantic.com",
          "first_name": "Allene",
          "gender": "Female",
          "id": 684,
          "last_name": "Bryenton",
        },
        Object {
          "Company": "Flashspan",
          "Title": "Senior Editor",
          "email": "amatthewkr@ning.com",
          "first_name": "Allene",
          "gender": "Female",
          "id": 748,
          "last_name": "Matthew",
        },
        Object {
          "Company": "Brightbean",
          "Title": "Financial Analyst",
          "email": "abickersl8@state.gov",
          "first_name": "Allegra",
          "gender": "Female",
          "id": 765,
          "last_name": "Bickers",
        },
        Object {
          "Company": "Tagchat",
          "Title": "Recruiting Manager",
          "email": "agarbettlr@naver.com",
          "first_name": "Allyn",
          "gender": "Female",
          "id": 784,
          "last_name": "Garbett",
        },
        Object {
          "Company": "Quaxo",
          "Title": "Nuclear Power Engineer",
          "email": "agonetpr@ocn.ne.jp",
          "first_name": "Allene",
          "gender": "Female",
          "id": 928,
          "last_name": "Gonet",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Alley
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Allen
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Allene
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Allene
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Allegra
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Allyn
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Allene
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
